src/environments/cd.js

```diff
@@ -0,0 +1,262 @@
+// @flow
+import buildCommon from "../buildCommon";
+import defineFunction from "../defineFunction";
+import mathMLTree from "../mathMLTree";
+import * as html from "../buildHTML";
+import * as mml from "../buildMathML";
+import {assertSymbolNodeType} from "../parseNode";
+import ParseError from "../ParseError";
+
+import type Parser from "../Parser";
+import type {ParseNode, AnyParseNode} from "../parseNode";
+
+const cdArrow = {
+    ">": "\\\\cdrightarrow",
+    "<": "\\\\cdleftarrow",
+    "=": "\\\\cdlongequal",
+    "A": "\\uparrow",
+    "V": "\\downarrow",
+    "|": "\\Vert",
+    ".": "no arrow",
+};
+
+const newCell = () => {
+    return {type: "styling", body: [], mode: "math", style: "display"};
+};
+
+const isLabelEnd = (node: AnyParseNode, endChar: string): boolean => {
+    return ((node.type === "mathord" || node.type === "atom") &&
+        node.text === endChar);
+};
+
+export function parseCD(parser: Parser): ParseNode<"array"> {
+    // Get the array's parse nodes with \\ temporarily mapped to \cr.
+    const parseNodes = [];
+    parser.gullet.beginGroup();
+    parser.gullet.macros.set("\\cr", "\\\\\\relax");
```

Comment:
Yes, `beginGroup` starts a new group which the definition will be limited to (because [the last argument of `set` defaults to `false`](https://github.com/KaTeX/KaTeX/blob/8c08a3c8a9e886585d1d86fa612f29e2787b195b/src/Namespace.js#L91)). A similar definition is made in the array environment.